sherlockml/clients/auth.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2018 ASI Data Science
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from datetime import datetime, timedelta
+import pytz
+from collections import namedtuple
+
+import requests
+
+
+AccessToken = namedtuple('AccessToken', ['token', 'expires_at'])
+
+
+class HudsonClient(object):
```

Comment:
(I discussed this in person with @acroz, but am noting it here for the benefit of others).

`Hudson` is an internal reference that won't make sense to external consumers of this library. I would prefer to name this `AuthClient` or similar, to aid discoverability and understanding.

Reply:
Done - renamed to `AccessTokenClient`